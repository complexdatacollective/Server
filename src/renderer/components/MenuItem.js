import React from 'react';
import PropTypes from 'prop-types';

const MenuItem = ({ children, action, ...props }) => (
  <button {...props} onClick={action}>
    {children}
  </button>
);

MenuItem.propTypes = {
  action: PropTypes.func.isRequired,
  children: PropTypes.node,
};

MenuItem.defaultProps = {
  children: null,
};

export default MenuItem;
