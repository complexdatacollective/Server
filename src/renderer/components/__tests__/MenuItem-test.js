/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import MenuItem from '../MenuItem';

const mockProps = {
  action: () => {},
};

describe('<MenuItem />', () => {
  it('should render', () => {
    const subject = shallow((
      <MenuItem {...mockProps}>
        Foo
      </MenuItem>
    ));

    expect(subject).toMatchSnapshot();
  });
});
