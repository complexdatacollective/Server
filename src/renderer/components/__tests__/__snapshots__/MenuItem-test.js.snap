// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    action={[Function]}
  >
    Foo
  </MenuItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Foo",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Foo",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Foo",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Foo",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
